src/k8s/pkg/k8sd/app/hooks_join.go

```diff
@@ -181,8 +181,9 @@ func onPostJoin(s *state.State, initConfig map[string]string) error {
 		return fmt.Errorf("failed to create kubernetes client: %w", err)
 	}
 
-	if err := client.WaitApiServerReady(s.Context); err != nil {
-		return fmt.Errorf("kube-apiserver did not become ready: %w", err)
+	err = client.WaitApiServerReady(s.Context)
+	if err != nil {
+		return fmt.Errorf("k8s api server did not become ready in time: %w", err)
```

Comment:
why `k8s api server` here? `kube-apiserver` is the standard service name

```suggestion
		return fmt.Errorf("kube-apiserver did not become ready: %w", err)
```